x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_domains_logic.ts

```diff
@@ -0,0 +1,125 @@
+import { HttpLogic } from '../../../shared/http';
+import { updateMetaPageIndex } from '../../../shared/table_pagination';
+import { EngineLogic } from '../engine';
+
+import {
+  CrawlerData,
+  CrawlerDataFromServer,
+  CrawlerDomain,
+  CrawlerDomainFromServer,
+} from './types';
+import { crawlerDataServerToClient, crawlerDomainServerToClient } from './utils';
+
+export interface CrawlerDomainsValues {
+  dataLoading: boolean;
+  domains: CrawlerDomain[];
+  meta: Meta;
+}
+
+interface CrawlerDomainsResponse {
+  results: CrawlerDomainFromServer[];
+  meta: Meta;
+}
+
+interface CrawlerDomainsActions {
+  deleteDomain(domain: CrawlerDomain): { domain: CrawlerDomain };
+  fetchCrawlerDomainsData(): void;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  onReceiveData(domains: CrawlerDomain[], meta: Meta): { domains: CrawlerDomain[]; meta: Meta };
+  crawlerDomainDeleted(data: CrawlerData): { data: CrawlerData };
+}
+
+export const CrawlerDomainsLogic = kea<MakeLogicType<CrawlerDomainsValues, CrawlerDomainsActions>>({
+  path: ['enterprise_search', 'app_search', 'crawler', 'crawler_domains_logic'],
+  actions: {
+    deleteDomain: (domain) => ({ domain }),
+    fetchCrawlerDomainsData: true,
+    onReceiveData: (domains, meta) => ({ domains, meta }),
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+    crawlerDomainDeleted: (data) => ({ data }),
+  },
+  reducers: {
+    dataLoading: [
+      true,
+      {
+        onReceiveData: () => false,
+        onPaginate: () => true,
+      },
+    ],
+    domains: [
+      [],
+      {
+        onReceiveData: (_, { domains }) => domains,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
+      {
+        onReceiveData: (_, { meta }) => meta,
+        onPaginate: (state, { newPageIndex }) => updateMetaPageIndex(state, newPageIndex),
+      },
+    ],
+  },
+  listeners: ({ actions, values }) => ({
+    fetchCrawlerDomainsData: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+      const { meta } = values;
+
+      const query = {
+        'page[current]': meta.page.current,
+        'page[size]': meta.page.size,
+      };
+
+      try {
+        const response = await http.get<CrawlerDomainsResponse>(
+          `/internal/app_search/engines/${engineName}/crawler/domains`,
+          {
+            query,
+          }
+        );
+
+        const domains = response.results.map(crawlerDomainServerToClient);
+
+        actions.onReceiveData(domains, response.meta);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+
+    deleteDomain: async ({ domain }) => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.delete<CrawlerDataFromServer>(
+          `/internal/app_search/engines/${engineName}/crawler/domains/${domain.id}`,
+          {
+            query: {
+              respond_with: 'crawler_details',
```

Comment:
I think it would be better to drop the `respond_with` param because we now call `actions.fetchCrawlerDomainsData()` after deletion.

We probably could do the same when creating a domain (in a follow-up PR):

https://github.com/elastic/kibana/blob/f2b9acf67b421a530e011d12aa9386c488d829b9/x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/add_domain_logic.ts#L262

Reply:
True.

Reply:
On the other hand, what's the benefit? Why not just leave it as is?

Reply:
We have a list of domains in CrawlerLogic, and as long as we're going to store that there we should keep it up to date. This way keeps it up to date. If we're not going to keep it up to date by fetching the crawler_details here then I think we should just remove that list of domains from state entirely.

Reply:
What does calling `actions.crawlerDomainDeleted(crawlerData)` do? It seems like it does nothing but I could be wrong.

Reply:
I see now. Sounds good to keep the `respond_with` param 👍